src/mon/MonClient.cc

```diff
@@ -111,14 +107,17 @@ int MonClient::get_monmap_privately()
 
   ldout(cct, 10) << "have " << monmap.epoch << " fsid " << monmap.fsid << dendl;
 
+  std::random_device rd;
```

Comment:
what happens to our reconnect attempts when std::random_device keeps returning the same sequence of monitors, such that the first mon_client_hunt_parallel of the chosen sequence are unreachable?

same issue in _add_conns()